src/main/java/com/rachelbock/resources/UserResource.java

```diff
@@ -75,20 +63,84 @@ public User getUserByUserName(@PathParam("user_name") String userName) {
      */
     @POST
     public User addUserToDatabase(NewUserRequest request) {
-        try (Connection conn = getConnection();
-             Statement stmt = conn.createStatement()){
+        try (Connection conn = ConnectionPool.getConnection();
+             Statement stmt = conn.createStatement()) {
 
             stmt.execute("INSERT INTO user_information (user_name, height, skill_level) VALUES " +
                     "('" + request.getUsername() + "', " + request.getHeight() + ", " + request.getSkill() + ")");
             return getUserByUserName(request.getUsername());
 
+        } catch (SQLIntegrityConstraintViolationException e) {
+            throw new BadRequestException("Username already exists");
         } catch (SQLException e1) {
             e1.printStackTrace();
         }
-
         throw new InternalServerErrorException("Could not create new user " + request.getUsername());
     }
 
+
+    public static final String RECOMMENDATIONS_QUERY = "SELECT * FROM climbs\n" +
+            "INNER JOIN user_information\n" +
+            "ON climbs.gym_rating = user_information.skill_level\n" +
```

Comment:
I'm actually not getting the same results using that syntax. I am pretty confident that I didn't just type it in wrong. For now I am going to leave it the same. Go ahead and merge